src/Core/Localization/Currency.php

```diff
@@ -137,7 +137,7 @@ public function __construct(
         $this->isoCode = $isoCode;
         $this->numericIsoCode = $numericIsoCode;
         $this->symbols = $symbols;
-        $this->precision = $precision;
+        $this->precision = $precision ? (int) $precision : null;
```

Comment:
```suggestion
        $this->precision = (int) $precision;
```
This should be enough